Makefile

```diff
@@ -27,10 +29,21 @@ APPNAME=tahoe-lafs
 default:
 	@echo "no default target"
 
+.PHONY: install-vcs-hooks
+## Install the VCS hooks to run linters on commit and all tests on push
+install-vcs-hooks: .git/hooks/pre-commit .git/hooks/pre-push
+
 .PHONY: test
 ## Run all tests and code reports
-test: .tox
-	tox -p auto
+test: .tox/create-venvs.log
+# Run codechecks first since it takes the least time to report issues early.
+	tox --develop -e codechecks
+# Run all the test environments in parallel to reduce run-time
+	tox --develop -p auto -e 'py27,py36,pypy27'
+.PHONY: test-py3-all
+## Run all tests under Python 3
+test-py3-all: .tox/create-venvs.log
```

Comment:
Why does anything depend on some logs file? (e.g. why not `tox.ini` or so?)

Reply:
Does [the commit message for the commit that made this change](https://github.com/tahoe-lafs/tahoe-lafs/pull/830/commits/b8268500effe48c52a7b84c12acce2424a92e70c) clarify?

Reply:
Since that will be lost I think something similar should be noted in the Makefile itself about the setup here (that is, about "why" there's a `.tox/create-venvs.log` -- which I at first assumed was a tox thing).

Reply:
The problem is that it's a very common make pattern where there's no clear single resulting build artifact and/or the timestamps of artifacts can't be counted on to really reflect what needs to be updated.  If you have no such build artifact then create one by writing to a log file with `tee`, so I wouldn't think it would be appropriate to comment at this one instance of that pattern.  IOW, such an explanation belongs in some sort of general make tips.